pilot/pkg/xds/ep_filters.go

```diff
@@ -119,7 +119,7 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 					},
 				}
 				// TODO: figure out a way to extract locality data from the gateway public endpoints in meshNetworks
-				gwEp.Metadata = util.BuildLbEndpointMetadata(network, model.IstioMutualTLSModeLabel, "", "", labels.Instance{})
+				gwEp.Metadata = util.BuildLbEndpointMetadata(network, model.IstioMutualTLSModeLabel, "", "", "", labels.Instance{})
```

Comment:
b.clusterID should work here.. it's easy to mix that with clusterName which is the envoy cluster.

Reply:
Thanks! Done.